text/0000-upgrade-procedures

```diff
@@ -0,0 +1,56 @@
+# The OpenTelemetry approach to upgrading
+
+Managing widely distributed software at scale requires careful design related to backwards compatibility, versioning, and upgrading. The OpenTelemetry approach is described below. If you are planning on using OpenTelemetry, it can be helpful to understand how we approach this problem.
+
+## Component Overview
+
+To facilitate smooth upgrading and long term support, OpenTelemetry clients are factored into several components. We use the following terms in the rest of this document.
+
+Packages is a generic term for units of code which reference each other via some form of dependency management. Note that every programming language has a different approach to dependency management, and may use a different term, such as module or library, to represent this concept. 
+
+The API refers to the set of software packages that contain all of the interfaces and constants needed to write OpenTelemetry instrumentation. An implementation of the API may be registered during application startup. If no other implementation is registered, the API registers a no-op implementation by default.
+
+The SDK refers to a framework which implements the API, provided by the OpenTelemetry project. While alternative API implementations may be written to handle special cases, we expect most users to install the SDK when running OpenTelemetry.
+
+Plugin Interfaces refer to extension points provided by the SDK. These include interfaces for controlling sampling, exporting data, and various other lifecycle hooks. Note that these interfaces are not part of the API. They are part of the SDK.
+
+Instrumentation refers to any code which calls the API. This includes the instrumentation provided by the OpenTelemetry project, third party instrumentation, plus application code and libraries which instrument themselves natively.
```

Comment:
nit: "Instrumentation refers to any code which calls the API to produce instrumentation signals".  (albeit non-instrumentation usage of the API is limited to SDK or SDK replacements)

Reply:
So, in this doc I want to define the term Instrumentation to specify _any_ caller of the API package, vs any implementor of the package interfaces. In other words, focus on the directionality of the dependency and not the use case. The relevant information for the reader is to understand that directionality, not the use case. I will try to make it super clear that the term "instrumentation" cover other random use cases as well, as far as this document is concerned.

Reply:
I guess @jsuereth was saying "SDK is not an instrumentation" (although SDK calls the API)?

Reply:
@jsuereth want to follow up on this when I add it to the spec?